### PR TITLE
Update dependencies and Java 8 support

### PR DESCRIPTION
**Pull request by lsanotes**

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -21,7 +21,7 @@ subprojects {
 }
 
 ext {
-    minSdkVersion = 9
+    minSdkVersion = 14
     compileSdkVersion = 28
-    sourceCompatibility = JavaVersion.VERSION_1_7
+    sourceCompatibility = JavaVersion.VERSION_1_8
 }
```
Comment (JakeWharton) [inline build.gradle:27]: I don't want to change this. The library is effectively done and won't benefit from any language feature of 8 in a meaningful way.

File: build.gradle [hunk 2]
```diff
@@ -21,7 +21,7 @@ subprojects {
...
 
 ext {
-    minSdkVersion = 9
+    minSdkVersion = 14
```
Comment (JakeWharton) [inline build.gradle:24]: Similarly, there doesn't seem to be a reason to change this

Comment (lsanotes) [reply build.gradle:24]: This change is caused by upgrading robolectric to 4.3. It integrated the androidx.test framework, which requires min sdk 14.

Comment (lsanotes) [reply build.gradle:24]: http://robolectric.org/migrating/

**Comment by lsanotes** [comment]
I'll split the pull request to smaller ones.

**Pull request closed without merging**